packages/shared/src/loadClerkJsScript.ts

```diff
@@ -144,20 +184,29 @@ const loadClerkJsScript = async (opts?: LoadClerkJsScriptOptions): Promise<HTMLS
     return null;
   }
 
-  const existingScript = document.querySelector<HTMLScriptElement>('script[data-clerk-js-script]');
-
-  if (existingScript) {
-    return waitForClerkWithTimeout(timeout);
-  }
-
   if (!opts?.publishableKey) {
     errorThrower.throwMissingPublishableKeyError();
     return null;
   }
 
+  const scriptUrl = clerkJsScriptUrl(opts);
+  const existingScript = document.querySelector<HTMLScriptElement>('script[data-clerk-js-script]');
+
+  if (existingScript) {
+    if (hasScriptRequestError(scriptUrl)) {
+      existingScript.remove();
+    } else {
+      try {
+        return await waitForClerkWithTimeout(timeout);
```

Comment:
I'm having seconds thoughts about this particular `return` here. I checked the code and it seems that there's no retry logic in IsomorphicClerk (the calling site) or in `loadClerkJsScript` - the retry logic is in `loadScript` which will not be called if we return here.

So, if I understand correctly, if the `<script>` fails to load and we're going to remove it and implicitly retry via `loadScript`. If, however, the script hasn't errored at the point of the check, we're going to wait for a timeout and then completely fail. This is good because the `status` will change, but it feels like we  should at least trigger a few retries first before failing completely. Failing to load the script cannot be recovered from the perspective of the user/dev once `status` switches so retrying makes sense

Reply:
I haven't confirmed it yet, but what you're saying makes sense! The current implementation just bails and does not retry if the existing script fails to load, your fix is a nice improvement! 

The `await` part is tricky though, if someone accidentally removes the keyword, retrying for this case will break and TS won't catch it as the return signature will not change (still a promise). Switching to `then().catch` might be a good first step towards making this a bit harder to break?

Another idea would be to move the retry logic from `loadScript` on level up (inside `loadClerkJsScript`) so its a little easier to understand who's responsible to handle the error thrown, but Im not 100% sold on that idea yet